tests/rptest/services/redpanda.py

```diff
@@ -1006,12 +1006,21 @@ def raise_on_bad_logs(self, allow_list=None):
                 f"Scanning node {node.account.hostname} log for errors...")
 
             match_errors = "-e ERROR" if self._raise_on_errors else ""
-            for line in node.account.ssh_capture(
-                    f"grep {match_errors} -e Segmentation\ fault -e [Aa]ssert {RedpandaService.STDOUT_STDERR_CAPTURE} || true"
-            ):
+            cmd = f"grep {match_errors} -e Segmentation\ fault -e [Aa]ssert {RedpandaService.STDOUT_STDERR_CAPTURE}"
+
+            # Account for grep exit code 1 on dedicated nodes such as EC2 instances
+            if self._dedicated_nodes:
+                cmd = cmd + ' || true'
+
+            for line in node.account.ssh_capture(cmd):
                 line = line.strip()
 
-                assert "No such file or directory" not in line
+                # Raise if the log file does not exists on dedicated nodes
+                # such as EC2 instances
+                if self._dedicated_nodes and 'No such file or directory' in line:
```

Comment:
I don't think this is necessary: raise_on_bad_logs is _only_ called on successful tests.  If the test succeeded and there is a redpanda node with no redpanda log then something else is broken (a redpanda cluster with more nodes than were started?)

Reply:
Alright, I'll drop this commit then.